include/sycl/algorithm/find.hpp

```diff
@@ -38,16 +38,20 @@
 #include <sycl/helpers/sycl_buffers.hpp>
 #include <sycl/helpers/sycl_namegen.hpp>
 #include <sycl/algorithm/algorithm_composite_patterns.hpp>
+#include <sycl/algorithm/buffer_algorithms.hpp>
 
 namespace sycl {
 namespace impl {
 
+#ifdef __COMPUTECPP__
+
 // a struct to store the result of a predicate comparison, and an index
 // we have to declare this here instead of in the function so that the sycl
 // kernel can see it
 typedef struct search_result {
   bool result;
   int index;
+  search_result() = default;
```

Comment:
on line 73, an empty buffer of type search_result is created, without a default constructor for search_result, the creation of this buffer is not possible in a C++ friendly way .